components/proxy/src/main/kotlin/com/hotels/styx/services/OriginsAdminHandler.kt

```diff
@@ -130,6 +135,16 @@ internal class OriginsAdminHandler(
         }
     }
 
+    private fun hasActiveHealthCheck(origin: RoutingObjectRecord) : Boolean {
+        val appId = lbGroupTag.find(origin.tags) ?: return false
+        serviceDb.entrySet().firstOrNull { (_, provider) ->
+            provider.type == HEALTH_CHECK_MONITOR
+                    && provider.tags.contains(targetTag(appId))
+                    && (provider.styxService as HealthCheckMonitoringService).isRunning()
+        } ?: return false
+        return true
+    }
```

Comment:
I would use more "functional" approach:

* This eliminates three `return` statements
* Therefore might be bit easier to follow?
* Bit more idiomatic in relation to our existing Kotlin codebase?

```
private fun hasActiveHealthCheck2(origin: RoutingObjectRecord) = lbGroupTag
        .find(origin.tags)
        ?.let { appId ->
            serviceDb.entrySet()
                    .firstOrNull { (_, provider) ->
                        provider.type == HEALTH_CHECK_MONITOR
                                && provider.tags.contains(targetTag(appId))
                                && (provider.styxService as HealthCheckMonitoringService).isRunning()
                    }
        } == null
```

But this is just my personal preference.

Reply:
Here is some more info about Kotlin scope functions: https://kotlinlang.org/docs/reference/scope-functions.html